### DISCUSSION TITLE
Is it possible to run the CLI using archived NWP data?

### DISCUSSION THREAD

**Discussion by DanielSWx** [firelab/windninja#504] [Q&A]
Hi,

I've been working with the CLI files for the past couple of months now. It's been great using it in real-time; however, I was wondering if it's possible to run it on archived use cases? I have a couple of use cases from 2017 I want to run using HRRR data. I've been scouring through the CLI documentation, but have been having trouble determining if this is feasible or not? Would I need to download the GRIB files from an archive server and then subset the data to the domain before running? Or is it possible to set arguments in the _.cfg_ file so that it downloads the data/subsets it automatically? I've tried setting the _forecast_time_ argument, but it tells me that there is no file available. I'm assuming that's because it's trying to connect to the NOMADS server, which doesn't store archived data past a few days if I'm not mistaken? Any and all guidance is greatly appreciated!

Thanks for all you guys do!

**Comment by latwood** [comment]
It's on the to-do list to prepare an archived HRRR downloader (hopefully by
this March?). I'm not sure if the code is still configured to handle
predownloaded and preprocessed older HRRR files or not, methods for
handling older HRRR from NOMADS back before updating WindNinja code for
later HRRR version changes should still be in the GitHub history. Last I
checked this, it seemed like it was just configured for the latest HRRR,
replacing the older HRRR methods with the whatever was latest at a given
time, but I'm not sure. Many of the other weather forecasts are setup to
handle both new and older versions of the same file types.

If WindNinja storage is still configured to work with the older HRRR date,
you should just be able to place the hrrr grib2 file in the same place as
the dem and use standard cli arguments as you would for running WindNinja
on a recently downloaded standard HRRR file.


But yes, part of the process of developing an archived HRRR downloader will
likely include using data from a different server, subsetting the data to
an area at least as big as the domain before running.

WindNinja should only connect to the NOMADS server when downloading a new
file, it just skips ahead to the internal file reading/processing and
storage steps if using a predownloaded file.


Hope this helps, usually I wait and let Natalie do the explaining but she's
out off and on for the next few weeks.

Loren

**Reply by DanielSWx** [reply]
This does, I sincerely appreciate your response!

**Comment by nwagenbrenner** [comment]
Sorry for my slow reply, I've been traveling. You should be able to
initialize with archived HRRR forecasts. You can get them from Google Cloud
Platform. As Loren mentioned, WindNinja doesn't have the ability to
download these historical forecasts yet. Also note that WindNinja can be
initialized with any NetCDF WRF output. You can see the Weather Model
Initialization tutorial that comes with the WindNinja installation  for
more details.

**Reply by DanielSWx** [reply]
Ah, understood now. Thank you!!
